.azure-pipelines-templates/checks.yml

```diff
@@ -9,5 +9,5 @@ jobs:
         clean: true
         fetchDepth: 1
 
-      - script: python3.8 scripts/extract-release-notes.py --target-git-version
+      - script: python3.10 scripts/extract-release-notes.py --target-git-version
```

Comment:
I think we ought to switch to calling `python3` everywhere, and communicate the version requirement through https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#python-requires

Wherever we can, using a shebang line or setuptools entry points would be nicer too, to avoid having to specify which  python, now that 2.x is thoroughly obsolete.